### PR TITLE
fix(error-handling): Use addError instead of throw

### PR DESCRIPTION
**Pull request by lkipke**
# Change Summary

This calls `interpreter.addError` instead of `throw new BrsError` in several spots so that we get these runtime errors printed out to the console when they occur. Also, it removes a `finally` block so that we are explicitly swallowing errors, rather than doing it implicitly.

### CODE REVIEW

File: test/e2e/Errors.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    });
+
+    test("components/errors/indexed-get.brs", async () => {
+        await execute([resourceFile("components", "errors", "indexed-get.brs")], outputStreams);
+
+        let errOutput = allArgs(outputStreams.stderr.write).filter((arg) => arg !== "\n");
+        expect(
+            errOutput[0].includes("Attempting to retrieve property from non-iterable value")
+        ).toBeTruthy();
+    });
```
Comment (alimnios72) [inline test/e2e/Errors.test.js:36]: Can we add a test case for `"Attempting to retrieve property from iterable with illegal index type"` error?

**Review by alimnios72 — COMMENTED** [review]
Looks good, just one small request

**Review by alimnios72 — APPROVED** [review]
Thanks, this looks great

**Review by sjbarag — APPROVED** [review]
Thanks for fixing this!🎉

**Pull request merged**